Removed unused prowjob scenario docs.

dekiel opened the pull request:
Removed limited tags on pr images notice.

**Related issue(s)**
See #10415

Sawthis reviewed (changes requested):
This PR contains only formatting changes.

dekiel commented:
Added missing changes. Added more details about default tags generated by the tool. Users duplicates tags by adding custom tags with value the same as default tag.

IwonaLanger commented on cmd/image-builder/README.md:
```diff
@@ -119,6 +116,29 @@ See the accepted inputs description in the [image-builder reusable workflow](/.g
+Image Builder provides a default tag for built images.
+The default tag is computed based on the template provided in the Image Builder configuration file.
+The default tag is always added to the image, even if the user provides custom tags.
+The Image Builder supports two default tags:
```
```suggestion
Image Builder supports two default tags:
```

IwonaLanger commented on cmd/image-builder/README.md:
```diff
@@ -119,6 +116,29 @@ See the accepted inputs description in the [image-builder reusable workflow](/.g
+
+If the name is not provided, it is evaluated from the value:
+
+- if the value is a string, it is used as a name directly. For example,`-tag latest` is equal to `-tag latest=latest`
+- if the value is go-template, it will be converted to a valid name. For example, `-tag v{{ .ShortSHA }}-{{ .Date }}` is equal
```
```suggestion
- If the value is a string, it is used directly as a name. For example,`-tag latest` is equal to `-tag latest=latest`
- If the value is go-template, it is converted to a valid name. For example, `-tag v{{ .ShortSHA }}-{{ .Date }}` is equal
```
